teps/NNNN-catalog-testing.md

```diff
@@ -0,0 +1,495 @@
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes two things:
+
+1. The interface that folks submitting `Tasks` to the Tekton Catalog will
+   need to comply with so that they can provide tests that we can use to
+   verifiy them; without these tests, `Tasks` will be at `community` level
+   only.
+2. The design of the (Tekton based!) infrastructure that will run these tests.
+
+## Motivation
+
+### Goals
+
+If we want people to rely on items in the Tekton Catalog, we need to be able to
+make sure that they work. And if we want people to submit items to the Tekton
+Catalog, they need to know how to provide tests that make sure that they work,
+and they need to be provided with infrastructure that they can test against.
+
+### Non-Goals
+
+* This proposal focuses on testing of `Tasks` in the catalog; the test infrastructure
+  would need to be expanded to support the testing of `Pipelines` as well
+* Catalog folder structure defined in [Tekton Catalog Organization TEP](https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md)
+* Catalog support tiers defined in [Tekton Catalog Tiers](https://docs.google.com/document/d/1BClb6cHQkbSpnHS_OZkmQyDMrB4QX4E5JXxQ_G2er7M/edit#heading=h.mfg0tcb14ixk)
+  * TODO: perhaps another TEP for this?
+* Handling (or not) of images required by tasks discussed in [Tekton Catalog Images](https://docs.google.com/document/d/1tkrDWd4Vud0mk5xOiox9JHfg96tjiawrGi0aonA590w/edit)
+* Tekton component compatibility discussed in https://github.com/tektoncd/pipeline/issues/2588
+  and implemented via the label `tekton.dev/minVersion`
+
+
+## Requirements
+
+### Quality
+
+1. If a resource is in the catalog, a user should feel confident that it will work as advertised
+  a. Linting and configuration tests should be applied to all resources
+  b. Testing should be applied regularly to all resources in the catalog
+  c. Folks developing resources should be able to easily run these tests themselves
+  d. It must be possible to apply this testing to resources that rely on external services (e.g. S3, slack)
+  e. Eventually tests should be run against varied Tekton installations, e.g. Tekton on different clouds
+
+### Tekton org external contributions
+ 1. We want to have high quality resources that users can rely on, however we
+   also want to make it possible to for anyone to submit resources to the catalog
+   with very little barrier in order to encourage contributions
+    1. They may initially be true of the catalog and later be true of the hub and not the catalog
+  2. Users should be able to define their own catalogs as well, and use resources from their own
```

Comment:
I don't understand this one - catalog is just a git repo today right? If we're going to make it more "special" and integrated, that would probably be it's own TEP?